### PR TITLE
Close Button

### PR DESCRIPTION
**Pull request by slauyama**
I messed up the first close button pull request. This one should work better. I have modified the index file so that it should automatically use the closeButton. Hopefully this one works better.

### CODE REVIEW

File: js/leaflet-sidebar.js [hunk 1]
```diff
@@ -13,6 +18,12 @@ L.Control.Sidebar = L.Control.extend({
         if (L.Browser.touch)
             L.DomUtil.addClass(this._sidebar, 'leaflet-touch');
 
+        // Create close button and attach it if configured
+        if (this.options.closeButton === true) {
+            this._closeButton = L.DomUtil.create('a', 'sidebar-close-button u-display-none', this._sidebar);
+            this._closeButton.innerHTML = '&times;';
+        }
```
Comment (Turbo87) [inline js/leaflet-sidebar.js]: I think it makes more sense to attach the close button to the `sidebar-content` div instead. In that case you also wouldn't need the whole `u-display-none` workaround.

File: js/leaflet-sidebar.js [hunk 2]
```diff
@@ -116,11 +140,19 @@ L.Control.Sidebar = L.Control.extend({
...
+                L.DomUtil.addClass(this.getCloseButton(), 'u-display-none');
+            }
         }
 
         return this;
     },
 
+    getCloseButton: function() {
+        return this._closeButton;
+    },
```
Comment (Turbo87) [inline js/leaflet-sidebar.js]: any reason why the close button DOM node should be part of the public interface?

**Comment by Turbo87** [comment]
@slauyama @JackDougherty Sorry for taking so long! I finally had a bit of time to review this PR and added my comments in the changes.

File: js/leaflet-sidebar.js (hunk 1 above)
Comment (MuellerMatthew) [inline js/leaflet-sidebar.js]: You should also make sure you set this to the tabindex = 0 so that it is accessible. 

this._closeButton.tabIndex = 0;

**Comment by MuellerMatthew** [comment]
One Issue I have notice with the close button is that it propagates the click event to the location below the close button on mobile devices.  Usually this isn't a big deal because it sets the focus on the map but it can also mean that the buttons in the top right corner of the map are receiving the event when the sidebar is in full screen mode on smaller screens.

**Comment by Turbo87** [comment]
obsoleted by b2a05efb1fb92ec1d1b9a076900ad7ba6c468993

**Pull request closed without merging**